packages/expo-image-picker/ios/EXImagePicker/EXImagePicker.m

```diff
@@ -389,9 +393,16 @@ - (BOOL)tryCopyImage:(NSDictionary * _Nonnull)info path:(NSString *)toPath {
   return false;
 }
 
-- (void)handleVideoWithInfo:(NSDictionary * _Nonnull)info saveAt:(NSString *)directory updateResponse:(NSMutableDictionary *)response
+- (void)handleVideoWithInfo:(NSDictionary * _Nonnull)info
+                     saveAt:(NSString *)directory
+             updateResponse:(NSMutableDictionary *)response
+          completionHandler:(void (^)(void))completionHandler
 {
-  NSURL *videoURL = info[UIImagePickerControllerMediaURL];
+  NSURL *videoURL = info[UIImagePickerControllerMediaURL] ?: info[UIImagePickerControllerReferenceURL];
+  if (videoURL == nil) {
+    self.reject(@"E_COULDNT_OPEN_FILE", @"Couldn't open video", nil);
+    return;
```

Comment:
```suggestion
    // not calling completionHandler here, as it's only purpose is to resolve the promise and we rejected it right here
    return;
```

Reply:
~Shouldn't we also clear `self.resolve` and `self.reject` here?~

Ah, I see the previous instances are lost if we call two methods consecutively without awaiting… 😢